batch/testResultsCollector.js

```diff
@@ -2,39 +2,73 @@ import {XUnitParser} from './parsers/xUnitParser';
 import mongoose from 'mongoose';
 import {TestSuiteModel} from '../models/testSuiteModel';
 import {TestCaseModel} from '../models/testCaseModel';
-
+import {GithubHelper} from '../helpers/githubHelper';
 
 class TestResultsCollector {
 	constructor(configuration) {
 		this.configuration = configuration;
-		mongoose.connect(configuration.mongo.uri, configuration.mongo.options);
+	}
+
+	connect() {
+		mongoose.connect(this.configuration.mongo.uri, this.configuration.mongo.options);
+	}
+
+	get parser() {
+		return new XUnitParser(this.configuration);
+	}
+
+	get testCaseModel() {
+		return TestCaseModel;
+	}
+
+	get githubHelper() {
+		return new GithubHelper(this.configuration);
+	}
+
+	createNewTestSuite(parsedTestSuiteData) {
+		return new TestSuiteModel({
+			name: parsedTestSuiteData.name,
+			failures: parsedTestSuiteData.failures,
+			timestamp: parsedTestSuiteData.timestamp
```

Comment:
How is that different from `new TestSuiteModel(parsedTestSuiteData)`? In which case, why make this function an instance method?